mm2src/coins/z_coin/storage/walletdb/wasm/storage.rs

```diff
@@ -0,0 +1,1474 @@
+            ticker: ticker.to_string(),
+            params: consensus_params,
+        };
+
+        Ok(db)
+    }
+
+    #[allow(unused)]
+    async fn lock_db(&self) -> WalletDbRes<WalletDbInnerLocked<'_>> {
+        self.db
+            .get_or_initialize()
+            .await
+            .mm_err(|err| ZcoinStorageError::DbError(err.to_string()))
+    }
+
+    pub async fn is_tx_imported(&self, tx_id: TxId) -> MmResult<bool, ZcoinStorageError> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+
+        // Recall where we synced up to previously.
+        let tx_table = db_transaction.table::<WalletDbTransactionsTable>().await?;
+        let index_keys = MultiIndex::new(WalletDbTransactionsTable::TICKER_TXID_INDEX)
+            .with_value(&self.ticker)?
+            .with_value(tx_id.0.to_vec())?;
+        let maybe_tx = tx_table.get_items_by_multi_index(index_keys).await?;
+
+        if !maybe_tx.is_empty() {
+            Ok(true)
+        } else {
+            Ok(false)
+        }
+    }
+
+    pub(crate) async fn init_accounts_table(
+        &self,
+        extfvks: &[ExtendedFullViewingKey],
+    ) -> MmResult<(), ZcoinStorageError> {
+        let ticker = self.ticker.clone();
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let walletdb_account_table = db_transaction.table::<WalletDbAccountsTable>().await?;
+
+        // check if account exists
+        let maybe_min_account = walletdb_account_table
+            .cursor_builder()
+            .only("ticker", &ticker)?
+            .bound("height", 0u32, u32::MAX)
+            .open_cursor(WalletDbAccountsTable::TICKER_ACCOUNT_INDEX)
+            .await?
+            .next()
+            .await?;
+        if maybe_min_account.is_some() {
+            return MmError::err(ZcoinStorageError::TableNotEmpty(
+                "Account table is not empty".to_string(),
+            ));
+        }
+
+        // Insert accounts
+        for (account, extfvk) in extfvks.iter().enumerate() {
+            let account_int = num_to_bigint!(account)?;
+            // address
+            let address = extfvk.default_address().unwrap().1;
+            let address = encode_payment_address(self.params.hrp_sapling_payment_address(), &address);
+
+            let account = WalletDbAccountsTable {
+                account: account_int.clone(),
+                extfvk: encode_extended_full_viewing_key(self.params.hrp_sapling_extended_full_viewing_key(), extfvk),
+                address,
+                ticker: ticker.clone(),
+            };
+
+            let index_keys = MultiIndex::new(WalletDbAccountsTable::TICKER_ACCOUNT_INDEX)
+                .with_value(&ticker)?
+                .with_value(account_int)?;
+
+            walletdb_account_table
+                .replace_item_by_unique_multi_index(index_keys, &account)
```

Comment:
Question: why do you use replace item and not add? just in case?
There is Insert into command for this functionality in sql.

Reply:
actually got this advice from Sergey, as this ensures we will never duplicate value.. literally an insert too